src/databases/indexedDb/loadRecords.ts

```diff
@@ -121,16 +121,67 @@ export async function loadRecords(countUpdater: ((totalCount: number) => void) |
 
 	if (countUpdater) void updateCount(collectionForCount, countUpdater);
 
-	if (needsMemorySort && sort) {
-		let items: object[];
-
-		if (sort.desc) processedCollection = processedCollection.reverse();
-		items = await processedCollection.sortBy(sort.id);
+	// Determine if there are out-of-line keys
+	const primKeySchema = table.schema.primKey;
+	const keyPath = primKeySchema.keyPath;
+	const isOutboundKey = (keyPath as string | string[] | null) === null;
 
-		const start = pageIndex * pageSize;
-		const end = Math.min(items.length, start + pageSize);
+	if (needsMemorySort && sort) {
+		// Check if we need to handle out-of-line keys
+		if (isOutboundKey) {
+			// Capture both items and their primary keys
+			const itemsWithKeys: Array<{ item: any; __primaryKey__: any }> = [];
+			await processedCollection.each((item, cursor) => {
+				itemsWithKeys.push({ item, __primaryKey__: cursor.primaryKey });
+			});
+
+			// Sort in memory
+			itemsWithKeys.sort((a, b) => {
+				const aValue = a.item[sort.id];
+				const bValue = b.item[sort.id];
+
+				if (aValue === bValue) return 0;
+				if (aValue == null) return 1;
+				if (bValue == null) return -1;
+				const comparison = aValue < bValue ? -1 : 1;
+				return sort.desc ? -comparison : comparison;
+			});
+
+			// Paginate
+			const start = pageIndex * pageSize;
+			const end = Math.min(itemsWithKeys.length, start + pageSize);
+			const unslicedItems = itemsWithKeys.slice(start, end);
+
+			return unslicedItems.map((entry) => ({ ...entry.item, id: entry.__primaryKey__ }));
+		} else {
+			// In-line keys - use Dexie's sortBy
+			let items: object[];
+
+			if (sort.desc) processedCollection = processedCollection.reverse();
+			items = await processedCollection.sortBy(sort.id);
+
+			const start = pageIndex * pageSize;
+			const end = Math.min(items.length, start + pageSize);
+
+			return items.slice(start, end);
+		}
+	}
 
-		return items.slice(start, end);
+	if (isOutboundKey) {
```

Comment:
We have renamed this `const`.

```suggestion
	if (hasOutOfLineKeys) {
```